helixbass/tree-sitter-lint
# Enable storing nodes in rule state

In this PR:
- refactor the `tree-sitter-grep` callback pattern to make it easier to be able to allow rules to store things like `tree_sitter::Node`'s in their per-file run local state
- some other things supporting the ESLint-rules plugin eg more "token"-related helpers exposed on `QueryMatchContext`

To test:
Per added tests you should be able to do things like declare a per-file-run rule state field whose type is `Node<'a>` and successfully store tree-sitter AST nodes there

Based on `rule-test-output-none`
